### ISSUE TITLE
Parallel Optimizer

### ISSUE THREAD

**Issue by evmckinney9** [Pitt-JonesLab/slam_decomposition#8]
- [ ] For parallelization, create an optimizer worker thread.

In current version, training plot returns all training loss lists for plotting. In parallelized version, if threshold is met, stop other threads and only return the most successful.
